tools/adventure-pack/goodies/typescript/iterator.prototype.some/index.test.ts

```diff
@@ -0,0 +1,53 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "./index";
+
+describe("Iterator.prototype.some", () => {
+  it("should return true if one element passes a test and false if all elements fails the test", () => {
+    const callbackFn = (element: number) => element % 2 === 0;
+    expect([2, 4, 6, 8, 10].values().some(callbackFn)).toBe(true);
+    expect([2, 4, 6, 7, 8, 10].values().some(callbackFn)).toBe(true);
+    expect([1, 4, 6, 8, 10].values().some(callbackFn)).toBe(true);
+    expect([1, 3, 5, 7, 9].values().some(callbackFn)).toBe(false);
+  });
+
+  it("should return false for an empty iterator", () => {
+    expect([].values().some((element: number) => element > 5)).toBe(false);
+  });
+
+  it("can test Map entries()", () => {
+    const map = new Map([
+      ["H", 1],
+      ["e", -2],
+      ["L", 3],
+      ["l", 0],
+      ["O", 5],
+      ["W", -1],
+      ["O", 0],
+      ["r", -5],
+      ["L", 1],
+      ["D", 2],
+    ]);
+    const callbackFn = ([key, value]: [string, number]) =>
+      key === key.toUpperCase() && value > 0;
+    expect(map.entries().some(callbackFn)).toBe(true);
+  });
+
+  it("can test Set values()", () => {
+    const set = new Set([1, 1, 2, 3, 3, 4, 5, 5, 6]);
+    expect(set.values().some((element: number) => element > 0)).toBe(true);
+  });
+
+  it("can test a Generator object", () => {
+    const generator = function* (): Generator<number, void, undefined> {
+      yield 2;
+      yield 4;
+      yield 6;
+      yield 8;
+      yield 10;
+    };
+
+    expect(generator().some((element: number) => element % 2 === 0)).toBe(true);
```

Comment:
Does TypeScript complain if you don't add a type annotation to the lambda? Usually lambda types can be inferred.

Reply:
ack!